### PR TITLE
🎉 (admin) add indicator chart editor

### PR DESCRIPTION
**Pull request by sophiamersmann**
Adds an editor for indicator-level chart configs.

### Details

- The indicator chart editor
    - A small note in the Basic tab states that this is an indicator chart
    - The Inheritance tab lists all charts that currently inherit from the indicator or where inheritance might be enabled
    - The `dimensions` field can't be edited (since it can't be inherited)
    - The Debug tab lists all configs (the admin config, etl config and the merged config)
- The variable page (under `/variables/123456`) has a new "Partial Grapher config" section that shows the ETL and the admin version. It comes with edit/delete buttons for the admin-authored config.

We should probably also add a few more tests for the new `grapherConfigAdmin` APIs, but I ran out of time :(

Knitted with a hot needle!

### CODE REVIEW

**Comment by sophiamersmann** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3886?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3886** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3886?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3793** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3793?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3782** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3782?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3827](https://github.com/owid/owid-grapher/pull/3827) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3827?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3767** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3767?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3743](https://github.com/owid/owid-grapher/pull/3743) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3743?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3846** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3846?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sophiamersmann and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: adminSiteClient/VariableEditPage.tsx [hunk 1]
```diff
@@ -528,6 +516,54 @@ class VariableEditor extends React.Component<{
...
+                                <button
+                                    type="button"
+                                    className="btn btn-danger ml-2"
+                                    onClick={async () => {
+                                        await this.context.admin.rawRequest(
+                                            `/api/variables/${variable.id}/grapherConfigAdmin`,
+                                            undefined,
+                                            "DELETE"
+                                        )
+                                        window.location.reload()
```
Comment (sophiamersmann) [inline adminSiteClient/VariableEditPage.tsx:557]: While talking about this in our dev meeting, I suddenly realized that I had left this criminal line in here! I'll replace it with an optimistic update in the next few days... :)

**Review by danyx23 — CHANGES_REQUESTED** [review]
Wow, what a feat! Really nicely done. I am especially delighted about the tests you added, but the whole refactor is very nicely done. Kudos!

I read through the whole source code but on the review side I'm not entirely sure that I 100% thought through all code paths. Everything seems to make perfect sense but there are probably a few bugs still lurking.

Two examples:
- when you edit the title of a chart that has automatic time added then this is added over and over again (e.g. http://staging-site-inherit-indicator-settings/admin/charts/7562/edit)
- When you turn on and off inheritance it resets too much sometimes IMHO. For example:
  - Create a new chart
  - add gdp per capita (world bank, ppp, used ca. 150 times) - this one has a non-trivial etl config
  - enable has map tab
  - save draft
  - enable inheritance
  - disable inheritance
  - has map tab is (wrongly IMHO) reset to false

For the last one I'be curious to know if chatgpt can write a regression test. If you show it the example of your current inherit on/off test and then give it the outline above (with the right setting name), can it quickly produce a valid test? Could be very useful for us. I was tempted to test it myself but don't have enough time today and didn't want to delay this review because of it.

When you have fixed the above I'd do another round of user testing but if things look good then then I think we could merge this soon and iron out any remaining issues as users find them.

**Comment by danyx23** [comment]
Hey @sophiamersmann, the Github Pull Requests Vs code extension mis-attributed the above review to this PR instead of the one below. I hope this doesn't cause any issues on your side - the things going wrong that I described apply to the PR one level down already and should be fixed there. I'll now review this PR.

**Comment by sophiamersmann** [comment]
I'll move your comment to the other PR then, so that it doesn't get too confusing.

File: adminSiteClient/VariableEditPage.tsx (hunk 1 above)
Comment (danyx23) [reply adminSiteClient/VariableEditPage.tsx:557]: this seems to still be here :)

**Review by danyx23 — COMMENTED** [review]
Looks good! The FieldsRow on the variables page that shows the two partial configs next to each other could benefit from an `aling-items: top` instead of center imho as the two boxes seem unbalanced in the example I looked at. 

I'm a little concerned about the perf of the queries to get all charts for an indciator since it joins on a view - but we can go ahead like this and monitor the perf and change it in a subequent pr if it is a problem.

**Review by danyx23 — APPROVED** [review]

File: db/model/Variable.ts [hunk 2]
```diff
@@ -356,7 +371,56 @@ export async function updateGrapherConfigAdminOfVariable(
...
+        SELECT
+            c.id AS chartId,
+            cc.full AS config,
+            cc.full ->> '$.isPublished' = "true" AS isPublished,
+            cxp.variableId = ? AS isChild,
+            c.isInheritanceEnabled
+        FROM charts c
+        JOIN chart_configs cc ON cc.id = c.configId
+        JOIN chart_dimensions cd ON cd.chartId = c.id
+        LEFT JOIN charts_x_parents cxp ON c.id = cxp.chartId
```
[the remark this answers is not in the document]
Comment (sophiamersmann) [reply db/model/Variable.ts:410]: Good point! I just checked where this query is used, and at the moment, this particular query is only used once in the indicator chart admin to compile a list of charts that inherit from that indicator. So, it's not a very critical path. Let's move forward then and come back to it when we need to :)

**Comment by sophiamersmann** [comment]
Changes are folded into its parent, #3793

**Pull request closed without merging**